File: src/modelo/Insumo.java
package modelo;

import java.util.ArrayList;
import java.util.List;

import javax.persistence.CascadeType;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.ManyToMany;

import dao.IDInterface;

@Entity
public class Insumo implements IDInterface {

	@Id
	@GeneratedValue (strategy=GenerationType.IDENTITY)
	private int id;
	private String nome;
	private boolean lactose;
	private boolean gluten;
	@ManyToMany (cascade=CascadeType.ALL)
	private List<Prato> pratos = new ArrayList<Prato>();
	

	public Insumo () {
		
	}
	
	public Insumo(String nome, boolean lactose, boolean gluten) {
		super();
		this.nome = nome.toUpperCase();
		this.lactose = lactose;
		this.gluten = gluten;
		this.pratos = new ArrayList <Prato>();
	}

	public int getId() {
		return id;
	}

	public void setId(int codinsumo) {
		this.id = codinsumo;
	}

	public String getNome() {
		return nome;
	}

	public void setNome(String nome) {
		this.nome = nome.toUpperCase();
	}

	public boolean isLactose() {
		return lactose;
	}

	public void setLactose(boolean lactose) {
		this.lactose = lactose;
	}

	public boolean isGluten() {
		return gluten;
	}

	public void setGluten(boolean gluten) {
		this.gluten = gluten;
	}
	
	public List<Prato> getPratos() {
		return this.pratos;
	}

	public void setPratos(List<Prato> pratos) {
		this.pratos = pratos;
	}

	@Override
	public String toString() {
		return nome;
	}
	
}

File: src/modelo/Insumo.php
<?php
namespace projeto\modelo;
use \projeto\DAO\IDInterface;

class Insumo implements IDInterface {
    private $lactose;
	private $gluten;
	use \projeto\traits\Nomeavel;
	
    public function __construct (String $nome, bool $lactose = false, bool $gluten = false) {
        $this->nome = strtoupper($nome);
        $this->lactose = $lactose;
        $this->gluten = $gluten;
    }

	public function getId() : int {
		return $id;
	}

	public function setId(int $id) {
		$this->id = $id;
	}

	public function getNome() : String{
		return $this->nome;
	}

	public function setNome(String $nome) {
		$this->nome = strtoupper($nome);
	}

	public function isLactose() : bool {
		return $lactose;
	}

	public function setLactose(bool $lactose) {
		$this->lactose = $lactose;
	}

	public function isGluten() : bool {
		return $this->gluten;
	}

	public function setGluten (bool $gluten) {
		$this->gluten = $gluten;
	}

	public function __toString()
	{
		return 'Nome: '.$this->getNome().'Possui lactose? '.($this->lactose ? 'SIM' : 'NÃO').'Possui glútem? '.($this->gluten ? 'SIM' : 'NÃO');
	}
}


?>